dace/codegen/targets/xilinx.py

```diff
@@ -515,13 +544,24 @@ def generate_host_function_body(self, sdfg, state, kernel_name, parameters,
         kernel_function_name = kernel_name
         kernel_file_name = "{}.xclbin".format(kernel_name)
 
+        rtl_starts = []
+        rtl_waits = []
+        for rtl_tasklet in rtl_tasklets:
+            rtl_starts.append(
+                f"  auto kernel_{rtl_tasklet} = program.MakeKernel(\"{rtl_tasklet}_top\"{', '.join([''] + [name for _, name, p, _ in parameters if isinstance(p, dace.data.Scalar)])}).ExecuteTaskFork();"
+            )
+            rtl_waits.append(f"  kernel_{rtl_tasklet}.wait();")
```

Comment:
Shouldn't the waits occur after all kernels have been launched?

Reply:
The waits are issued after launching the kernels:
```
        kernel_stream.write(
            """\
  {rtl_starts}
  auto kernel = program.MakeKernel({kernel_function_name}, "{kernel_function_name}", {kernel_args});
  const std::pair<double, double> elapsed = kernel.ExecuteTask();
  {rtl_waits}
  std::cout << "Kernel executed in " << elapsed.second << " seconds.\\n" << std::flush;
```
So the generated code will be
```
    auto kernel_rtl_tasklet_0_0_0 = program.MakeKernel("rtl_tasklet_0_0_0_top").ExecuteTaskFork();
    auto kernel = program.MakeKernel(device_state_0, "device_state_0", fpga_A, fpga_B, N);
    const std::pair<double, double> elapsed = kernel.ExecuteTask();
    kernel_rtl_tasklet_0_0_0.wait();
    std::cout << "Kernel executed in " << elapsed.second << " seconds.\n" << std::flush;
```